### PR TITLE
Add a layer opacity manager

### LINKED ISSUE #381: Activation of a layer vs making them invisible
There is a real UX problem:
If a layer is invisible and you deactivate it and activate it later on it stays invisible.
The user will never understand what's happening.

Restant c2c 01/04/2015 : 0.25j

### PR DESCRIPTION
**Pull request by elemoine**
This commit adds a layer opacity manager service responsible for resetting the opacity of layers added to the map.

Fixes #381.

### CODE REVIEW

File: geoportailv3/static/js/layeropacitymanagerservice.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+/**
+ * @fileoverview This file defines the "appLayerOpacityManger" service. This
```
Comment (pgiraud) [inline geoportailv3/static/js/layeropacitymanagerservice.js]: Missing "a" in "appLayerOpacityManger".

File: geoportailv3/static/js/layeropacitymanagerservice.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  goog.events.listen(layers, ol.CollectionEventType.ADD,
+      /**
+       * @param {ol.CollectionEventType} evt Collection event.
+       */
+      function(evt) {
+        var layer = /** @type {ol.layer.Layer} */ (evt.element);
+        var layerMetadata = layer.get('metadata');
+        var layerStartOpacity = goog.isDef(layerMetadata) &&
+            layerMetadata.hasOwnProperty('start_opacity') ?
+            +layerMetadata['start_opacity'] : 1;
```
Comment (pgiraud) [inline geoportailv3/static/js/layeropacitymanagerservice.js]: Can you explain me what the "+" is here for?

**Comment by pgiraud** [comment]
I added a small comment. Otherwise it looks good.

File: geoportailv3/static/js/layeropacitymanagerservice.js (hunk 2 above)
Comment (elemoine) [inline geoportailv3/static/js/layeropacitymanagerservice.js]: `start_opacity` is a string in the JSON document. `+` casts it to a number.

**Comment by elemoine** [comment]
> I added a small comment. Otherwise it looks good.

Thanks. I'll fix the typo and merge.

**Pull request merged**